statev2/src/replication/network.rs

```diff
@@ -0,0 +1,88 @@
+//! The network abstraction of the replication layer; defines the interface for
+//! sending and receiving raft messages from cluster peers
+
+use raft::prelude::Message as RaftMessage;
+
+use super::error::ReplicationError;
+
+/// The central trait that must be implemented by any networking layer used
+/// by the replication layer
+pub trait Network {
```

Comment:
nit: `Network` feels a bit vague, thoughts on renaming this to e.g. `ReplicationNetwork`?